whelk-core/src/main/groovy/whelk/search/XLQLQuery.java

```diff
@@ -587,11 +618,71 @@ private Map<String, Object> buildStats(
             }
         });
 
-        return Map.of(JsonLd.ID_KEY, "#stats",
-                "sliceByDimension", sliceByDimension);
+        return sliceByDimension;
+    }
+
+    private List<Map<String,Object>> getBoolFilters(Map<String, Object> statsRepr, SimpleQueryTree sqt, Map<String, String> nonQueryParams) throws InvalidQueryException {
+        var defaultBoolFilters = getDefaultBoolFilters(statsRepr);
+        var existing = sqt.getBoolFiltersByAlias();
+        List<Map<String, Object>> results = new ArrayList<>();
+
+        for (var entry : defaultBoolFilters.entrySet()) {
+            var alias = entry.getKey();
+            var filter = (SimpleQueryTree.Node) entry.getValue().get("filter");
+            var defaultStatus = (FilterStatus) entry.getValue().get("status");
+            var bfNode = existing.get(alias);
+
+            SimpleQueryTree newTree;
+            boolean isSelected;
+
+            if (bfNode == null) {
+                var newStatus = switch (defaultStatus) {
+                    case INACTIVE -> FilterStatus.ACTIVE;
+                    case ACTIVE -> FilterStatus.INACTIVE;
+                };
+                newTree = sqt.andExtend(new SimpleQueryTree.BoolFilter(alias, newStatus, filter));
+                isSelected = false;
+            } else {
+                newTree = sqt.excludeFromTree(bfNode);
+                isSelected = true;
+            }
+
+            Map<String, Object> res = new LinkedHashMap<>();
+            // TODO: fix form
+            res.put("_selectHeader", entry.getValue().get("selectHeader"));
+            res.put("view", Map.of(JsonLd.ID_KEY, makeFindUrl(newTree, makeParams(nonQueryParams))));
+            res.put("_selected", isSelected);
+
+            results.add(res);
+        }
+
+        return results;
+    }
+
+    public Map<String, Map<String, Object>> getDefaultBoolFilters(Map<String, Object> statsRepr) throws InvalidQueryException {
+        // TODO: get from apps.jsonld
+        List<Map<String, String>> boolFilters = List.of(
+                Map.of("alias", "excludeEplikt", "filter", "NOT bibliography:\"sigel:EPLK\"", "status", "active", "selectHeader", "Inkludera e-plikt"),
+                Map.of("alias", "excludePreliminary", "filter", "NOT encodingLevel:(\"marc:PartialPreliminaryLevel\" OR \"marc:PrepublicationLevel\")", "status", "active", "selectHeader", "Inkludera kommande publiceringar"),
+//                Map.of("alias", "availableOnline", "filter", "_TODO", "status", "inactive", "selectHeader", "Endast material som finns online"),
+//                Map.of("alias", "isDigital", "filter", "_TODO", "status", "inactive", "selectHeader", "Endast digitalt material"),
+                Map.of("alias", "hasImage", "filter", "image:*", "status", "inactive", "selectHeader", "Endast resurser med bild")
```

Comment:
I had imagined this as to separate levels of abstraction.

something like
"defining the symbol"
`"alias": "excludeEplikt", "filter": "NOT bibliography:\"sigel:EPLK\""`

"specifying the site default filter"
filter: "rdf:type: Work excludeEplikt excludePreliminary"

"defining the bool filter options"
`"filter": "excludeEplikt" something something bool

now we limit it so that every aliased filter has to have a bool filter.
maybe that is fine.

`selectHeader` should either be a `prefLabelByLang` or a key that can beused for translation client side.

"Endast resurser med bild" is ambigous -> "Endast resurser med omslags-/miniatyrbild"

Reply:
selectHeader should either be a prefLabelByLang or a key that can beused for translation client side.

Reply:
"Endast resurser med bild" is ambigous -> "Endast resurser med omslags-/miniatyrbild"